pulp_smash/tests/rpm/api_v2/test_service_resiliency.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+"""Test the resiliency of Pulp's services.
+
+Pulp is designed to be resilient. It should be possible for a service to
+disappear and reappear, with the only consequence being slower processing of
+tasks. This module has tests for Pulp's resilience in the face of such issues.
+"""
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, cli, config, selectors, utils
+from pulp_smash.constants import (
+    PULP_SERVICES,
+    REPOSITORY_PATH,
+    RPM_UNSIGNED_FEED_URL,
+)
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+
+_PULP_WORKERS_CFG = '/etc/default/pulp_workers'
+
+
+class MissingWorkersTestCase(unittest.TestCase):
+    """Test that Pulp deals well with missing workers.
+
+    When executed, this test case will do the following:
+
+    1. Create several repositories.
+    2. Request that each repository be synchronized and published. Assuming
+       that there is only one worker, this queues several tasks for that one
+       worker.
+    3. Immediately restart the ``pulp_workers`` service. It's important that
+       ``pulp_workers`` be restarted, not started and stopped. Verify that the
+       syncs and publishes complete. (If the issue under test still affects
+       Pulp, the worker handling the syncs and publishes may have broken.)
+
+    This test targets `Pulp #2835 <https://pulp.plan.io/issues/2835>`_.
+    """
+
+    def setUp(self):
+        """Ensure there is only one pulp worker."""
+        self.cfg = config.get_config()
+        if selectors.bug_is_untestable(2835, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/2835')
+        sudo = '' if utils.is_root(self.cfg) else 'sudo'
+        cli.Client(self.cfg).machine.session().run(
+            "{} bash -c 'echo PULP_CONCURRENCY=1 >> {}'"
+            .format(sudo, _PULP_WORKERS_CFG)
+        )
+        cli.GlobalServiceManager(self.cfg).restart(PULP_SERVICES)
+
+    def test_all(self):
+        """Test that Pulp deals well with missing workers."""
+        # Create a repository. No repository addCleanup is necessary, because
+        # Pulp will be reset after this test.
+        client = api.Client(self.cfg, api.json_handler)
+        repos = []
+        for _ in range(5):
+            body = gen_repo()
+            body['importer_config']['feed_url'] = RPM_UNSIGNED_FEED_URL
+            body['distributors'] = [gen_distributor()]
+            repo = client.post(REPOSITORY_PATH, body)
+            repos.append(client.get(repo['_href'], params={'details': True}))
+
+        # Request that the repo be synchronized and published. Don't wait.
+        client.response_handler = api.code_handler
+        responses = []
+        for repo in repos:
+            responses.append(client.post(
+                urljoin(repo['_href'], 'actions/sync/')
+            ))
+            responses.append(client.post(
```

Comment:
Calling a publish is fine but not necessary. Having at least one task type per repo is the important part.